lib/plugins/aws/package/compile/events/apiGateway/lib/requestValidator.test.js

```diff
@@ -0,0 +1,449 @@
+'use strict';
+
+const chai = require('chai');
+const runServerless = require('../../../../../../../../tests/utils/run-serverless');
+const fixtures = require('../../../../../../../../tests/fixtures');
+
+const expect = chai.expect;
+chai.use(require('chai-as-promised'));
+
+describe('#compileRequestValidators()', () => {
+  after(fixtures.cleanup);
+
+  describe(' reusable schemas ', () => {
+    it('Should process schema from apiGateway provider, full config', () =>
+      fixtures.extend('requestSchema', {}).then(fixturePath =>
+        runServerless({
+          cwd: fixturePath,
+          cliArgs: ['package'],
+        }).then(serverless => {
+          const cfResources = serverless.service.provider.compiledCloudFormationTemplate.Resources;
+          const naming = serverless.getProvider('aws').naming;
+          const modelLogicalId = naming.getModelLogicalId('TestModel', 'any', 'application/json');
+          const modelResource = cfResources[modelLogicalId];
+          expect(modelResource).to.deep.equal({
+            Type: 'AWS::ApiGateway::Model',
+            Properties: {
+              ContentType: 'application/json',
+              Description: 'Test Description',
+              Name: 'TestModel',
+              RestApiId: {
+                Ref: 'ApiGatewayRestApi',
+              },
+              Schema: {
+                $schema: 'http://json-schema.org/draft-04/schema#',
+                definitions: {},
+                properties: {
+                  id: {
+                    pattern: '[0-9]+',
+                    title: 'ID for object',
+                    type: 'number',
+                  },
+                },
+                required: ['id'],
+                title: 'Test Validation Schema',
+                type: 'object',
+              },
+            },
+          });
+        })
+      ));
+
+    it('Should process schema from apiGateway provider, missing name and description', () =>
+      fixtures.extend('requestSchema', {}).then(fixturePath =>
```

Comment:
Let's reuse one `serverless` instance when we do not change fixture. We can put `serverless` instance initialization into `before` block, and then in `it`'s just confirm on generated template.

It'll run also way faster then